src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,305 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { NumberInputIncrement, NumberInputProps } from './types';
+import { merge } from '@utilities/merge';
+import { IconCross16, IconMinus16, IconPlus16 } from '@foundation/Icon';
+import { useMemoizedId } from '@hooks/useMemoizedId';
+import { Validation, validationClassMap } from '@utilities/validation';
+import { KeyboardEvent } from '@react-types/shared';
+import { GetStatusIcon } from '@utilities/input';
+import { FOCUS_WITHIN_STYLE } from '@utilities/focusStyle';
+
+const INCREMENT_KEYS = ['ArrowUp', 'ArrowRight'];
+const DECREMENT_KEYS = ['ArrowDown', 'ArrowLeft'];
+const SPECIAL_KEYS = ['Backspace', 'Tab'];
+type FocusedState = {
+    isFocused: boolean;
```

Comment:
@ryancarville Let's dedicate time for the isFocusVisible hook on a separate PR.

Reply:
Agree on the `IsFocusVisible` hook.  I can remove this functionality then and plan to re-introduce it once the hook it implemented.

Reply:
As for the a11y issues I will have a look again.  I did speak with @LidijaStoilova a few weeks ago about it and the `label` tag was the biggest concern.  I will re-visit the topic.